model-optimizer/extensions/front/tf/TFSliceToSlice.py

```diff
@@ -19,63 +19,48 @@
+        slice_name = tf_slice_node.soft_get('name', tf_slice_node.id)
         slice_node = Slice(graph).create_node()
-        rename_nodes([(node, slice_name + '/to_be_removed'), (slice_node, slice_name)])
+        rename_nodes([(tf_slice_node, slice_name + '/to_be_removed'), (slice_node, slice_name)])
+        add_node = Add(graph, {'name': slice_name + '/end_const'}).create_node()
 
-        eq_node = Equal(graph, {'name': slice_name + '/equal'}).create_node()
-        minus_one_node = Const(graph, {'name': slice_name + '/minus_one', 'value': np.array(-1)}).create_node()
-        int32_max_node = Const(graph, {'name': slice_name + '/int32_max', 'value': np.iinfo(np.int32).max}).create_node()
-        select_node = Select(graph, {'name': slice_name + '/select'}).create_node()
+        # reconnect begin and input from TFSlice to Slice
+        tf_slice_node.in_port(0).get_connection().set_destination(slice_node.in_port(0))
+        tf_slice_node.in_port(1).get_connection().set_destination(slice_node.in_port(1))
+        tf_slice_node.in_port(2).get_connection().set_destination(add_node.in_port(0))
+        slice_node.in_port(1).get_connection().add_destination(add_node.in_port(1))
 
-        # node to convert sizes to ends
-        sum_node = Add(graph, {'name': slice_name + '/end_const'}).create_node()
+        shapeof_node = Shape(graph, {'name': slice_name + '/ShapeOf'}).create_node()
+        slice_node.in_port(0).get_connection().add_destination(shapeof_node.in_port(0))
 
-        # reconnect input from tfslice to slice
-        node.in_port(0).get_source().connect(slice_node.in_port(0))
-        node.in_port(0).disconnect()
-        # reconnect begin of tfslice to start of slice
-        node.in_port(1).get_source().connect(slice_node.in_port(1))
-        node.in_port(1).disconnect()
+        # nodes to check if size == -1, if so take the whole range
+        eq_node = create_op_with_const_inputs(graph, Equal, {0: int64_array(-1)}, {'name': slice_name + '/equal'})
+        add_node.in_port(0).get_connection().add_destination(eq_node.in_port(1))
 
-        # (size -> ends) reconnect begins and sizes to sum to evaluate ends for Slice
-        # connects begins to slice
-        slice_node.in_port(1).get_source().connect(sum_node.in_port(0))
-        node.in_port(2).get_source().connect(sum_node.in_port(1))
-        node.in_port(2).disconnect()
-
-        # if size[i] == -1 when take int32_max as end[i]
-        sum_node.in_port(1).get_source().connect(eq_node.in_port(0))
-        minus_one_node.out_port(0).connect(eq_node.in_port(1))
-        # from equal to 0 port of select
-        eq_node.out_port(0).connect(select_node.in_port(0))
-        # from int32_max to 1 of select
-        int32_max_node.out_port(0).connect(select_node.in_port(1))
-        # from sum to 2nd of select
-        sum_node.out_port(0).connect(select_node.in_port(2))
-        # out of select to end (2nd of slice)
+        # select requires equal dtypes
```

Comment:
misleading comment.

Reply:
Why misleading? `Select` indeed requires the same dtypes and this is the reason why we add `Cast`